src/Library/demos/Column View/main.js

```diff
@@ -0,0 +1,91 @@
+import Gtk from "gi://Gtk";
+import Gio from "gi://Gio";
+
+const column_view = workbench.builder.get_object("column_view");
+const col1 = workbench.builder.get_object("col1");
+const col2 = workbench.builder.get_object("col2");
+const col3 = workbench.builder.get_object("col3");
+
+function createFirstCol() {
+  const factory = col1.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const label_widget = list_item.get_child();
+    const model_item = list_item.get_item();
+    label_widget.label = model_item.get_display_name();
+  });
+}
+
+function createSecCol() {
+  const factory = col2.factory;
+  factory.connect("setup", (factory, list_item) => {
+    const label = new Gtk.Label({
+      margin_start: 12,
+      margin_end: 12,
+    });
+    list_item.set_child(label);
+  });
+  factory.connect("bind", (factory, list_item) => {
+    const label_widget = list_item.get_child();
+    const model_item = list_item.get_item();
+    label_widget.label = model_item.get_size().toString();
```

Comment:
Probably we want to use a human readable size here, but let's leave that for the polish phase.

Reply:
Be sure to double-check my math, but this should be good enough:

```js
if (Math.floor(size / 1_000_000_000) > 0)
  label_widget.label = `${(size / 1_000_000_000).toFixed(1)} GB`
else if (Math.floor(size / 1_000_000) > 0)
  label_widget.label = `${(size / 1_000_000).toFixed(1)} MB`
else if (Math.floor(size / 1000) > 0)
  label_widget.label = `${(size / 1000).toFixed(1)} kB`
else
  label_widget.label = `${(size / 1000)} bytes`
```